src/ReadTransaction.ts

```diff
@@ -11,49 +14,52 @@ import { Transaction } from './Transaction'
 export class ReadTransaction extends Transaction<ReadTransactionRequest, ReadTransactionReply> {
   public async start() {
     const startReq = new StartTransactionRequest()
-    startReq.setStoreid(super.storeID)
-    startReq.setModelname(super.modelName)
+    startReq.setStoreid(this.storeID)
+    startReq.setModelname(this.modelName)
     const req = new ReadTransactionRequest()
     req.setStarttransactionrequest(startReq)
-    super.client.start()
-    super.client.send(req)
+    this.client.start()
+    this.client.send(req)
   }
 
   public async has(entityIDs: string[]) {
-    return new Promise<boolean>((resolve, reject) => {
+    return new Promise<ModelHasReply.AsObject>((resolve, reject) => {
       const hasReq = new ModelHasRequest()
       hasReq.setEntityidsList(entityIDs)
       const req = new ReadTransactionRequest()
       req.setModelhasrequest(hasReq)
       this.client.onMessage((message: ReadTransactionReply) => {
-        resolve(message.hasModelhasreply())
```

Comment:
We _could_ still check here for a reply, but I think its probably sufficient to return an optional here?